src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1105,6 +1149,27 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
 
                 addressTransactions.Add(newTransaction);
                 this.AddInputKeysLookupLocked(newTransaction);
+                
+                // Workaround for now to populate index as the address lookup doesn't give us all the requisite data
+                // TODO: Either amend address lookup to provide additional fields or optimise in some other way
+                bool finish = false;
+                foreach (Wallet wallet in this.Wallets)
+                {
+                    if (finish) break;
+                    foreach (HdAccount account in wallet.GetAccounts())
+                    {
+                        if (finish) break;
+                        foreach (HdAddress address1 in account.GetCombinedAddresses())
+                        {
+                            if (finish) break;
+                            if (address == address1)
+                            {
+                                this.AddTxLookupLocked(wallet, account, address, newTransaction);
```

Comment:
So just to understand the logic... If teh address associated to the script matches an address in the wallet we add it?

Can't we do some fancy LINQ statement using `SelectMany` here:

`var applicableAddresses = this.Wallets.SelectMany(w=w.GetAccounts()).SelectMany(a=>a.GetCombinedAddresses())`

Reply:
I suppose so, although it might make it somewhat unclear what is being done. Basically the only reason I loop through everything is so that we obtain knowledge of the `Wallet` and `HdAccount` instances to use (as we already know the `HdAddress` and `TransactionData`, and they have no link to the other data to allow them to be efficiently looked up). With a change to the data returned by `scriptToAddressLookup` this would be completely unnecessary, but I didn't want to change too much more in this PR.